sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -101,20 +101,19 @@ impl PgLTree {
     /// creates ltree from an iterator with checking labels
     pub fn from_iter<I, S>(labels: I) -> Result<Self, PgLTreeParseError>
     where
-        S: Into<String>,
+        String: From<S>,
         I: IntoIterator<Item = S>,
     {
         let mut ltree = Self::default();
         for label in labels {
-            ltree.push(&label.into())?;
+            ltree.push(String::from(label).parse()?);
         }
         Ok(ltree)
     }
 
     /// push a label to ltree
-    pub fn push(&mut self, label: &str) -> Result<(), PgLTreeParseError> {
-        self.labels.push(PgLTreeLabel::new(label)?);
-        Ok(())
+    pub fn push(&mut self, label: PgLTreeLabel) {
```

Comment:
This is unfortunately a breaking change, though, and will need to wait for 0.6.0. I don't have an exact timeline on that right now but I expect it to be the next release we do, probably in a month or so.

Reply:
Thanks for the info! The work-around is not too terrible anyways

Reply:
> Thanks for the info! The work-around is not too terrible anyways

You could add a new `push_` function with the new signature and leave the existing one as-is.